docs/builder-guides/XVM and WASM/interact_with_wasm_contract.md

```diff
@@ -0,0 +1,318 @@
+
+```js
+// the address we are going to query
+const target = '5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY';
+const from = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';
+
+// only 1 param needed, the actual address we are querying for (more
+// params can follow at the end, separated by , if needed by the message)
+const callValue = await contract.query.balanceOf(from, { value: 0, gasLimit: -1 }, target);
+```
+
+In this example we have specified a `gasLimit` of `-1`, in a subsequent section we will expand on this. for now, just remember that is indicated to use max available, i.e. we don't explicitly want to specify a value.
+
+When executing it encodes the message using the selector and the input values to allow execution in the contract environment. This can be executed on any contract message, unlike the examples that will follow below it will only read state, not actually execute and therefore not consume any real value from the account.
+
+An alternative for reading would be via the lower-level `.read` method, in this case
+
+```javascript
+// Perform the actual read (no params at the end, for the `get` message)
+// (We perform the send from an account address, it doesn't get executed)
+const callValue = await contract
+  .read('get', { value, gasLimit })
+  .send(alicePair.address);
+
+// The actual result from RPC as `ContractExecResult`
+...
+```
+
+In cases where the ABI messages have conflicting names, instead of the `'get'` string the actual message index (or message from the Abi itself) can be passed-through.
+
+
+## Sending a transaction
+
+In addition to using the `.query.<messageName>` on a contract, the `.tx.<messageName>` method is provides to send an actual encoded transaction to the contract, allow for execution and have this applied in a block. Expanding on our previous examples, we can now execute and then retrieve the subsequent value -
+
+```javascript
+// We will use these values for the execution
+const value = 0; // only useful on isPayable messages
+const gasLimit = 3000n * 1000000n;
+const incValue = 1;
+
+// Send the transaction, like elsewhere this is a normal extrinsic
+// with the same rules as applied in the API (As with the read example,
+// additional params, if required can follow - here only one is needed)
+await contract.tx
+  .inc({ value, gasLimit }, incValue)
+  .signAndSend(alicePair, (result) => {
+    if (result.status.isInBlock) {
+      console.log('in a block');
+    } else if (result.status.isFinalized) {
+      console.log('finalized');
+    }
+  });
+```
+
+If we perform the same `query.get` read on the value now, it would be `124`. For lower-level access, like we have in the `Blueprint` via `.createContract` you can also perform the execution via the `.exec` function, which would yield equivalent results -
```

Comment:
"we perform" vs "you can perform". Should be consistent throughout the document.